<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WgtComm</class>
 <widget class="QWidget" name="WgtComm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>793</width>
    <height>567</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" rowstretch="41,13" columnstretch="8,3">
   <property name="leftMargin">
    <number>10</number>
   </property>
   <property name="topMargin">
    <number>10</number>
   </property>
   <property name="rightMargin">
    <number>10</number>
   </property>
   <property name="bottomMargin">
    <number>10</number>
   </property>
   <property name="spacing">
    <number>10</number>
   </property>
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_3" stretch="5,0,1">
     <property name="spacing">
      <number>10</number>
     </property>
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="palette">
        <palette>
         <active>
          <colorrole role="WindowText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Button">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Light">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Midlight">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Dark">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>127</red>
             <green>127</green>
             <blue>127</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Mid">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>170</red>
             <green>170</green>
             <blue>170</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Text">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="BrightText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ButtonText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Base">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Window">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Shadow">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="AlternateBase">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ToolTipBase">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>220</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ToolTipText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
         </active>
         <inactive>
          <colorrole role="WindowText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Button">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Light">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Midlight">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Dark">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>127</red>
             <green>127</green>
             <blue>127</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Mid">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>170</red>
             <green>170</green>
             <blue>170</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Text">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="BrightText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ButtonText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Base">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Window">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Shadow">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="AlternateBase">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ToolTipBase">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>220</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ToolTipText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
         </inactive>
         <disabled>
          <colorrole role="WindowText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>127</red>
             <green>127</green>
             <blue>127</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Button">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Light">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Midlight">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Dark">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>127</red>
             <green>127</green>
             <blue>127</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Mid">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>170</red>
             <green>170</green>
             <blue>170</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Text">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>127</red>
             <green>127</green>
             <blue>127</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="BrightText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ButtonText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>127</red>
             <green>127</green>
             <blue>127</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Base">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Window">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="Shadow">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="AlternateBase">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>255</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ToolTipBase">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>255</red>
             <green>255</green>
             <blue>220</blue>
            </color>
           </brush>
          </colorrole>
          <colorrole role="ToolTipText">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>0</red>
             <green>0</green>
             <blue>0</blue>
            </color>
           </brush>
          </colorrole>
         </disabled>
        </palette>
       </property>
       <property name="widgetResizable">
        <bool>false</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>549</width>
          <height>289</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="WgtScroll" name="wgtScroll" native="true"/>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="txtTosend"/>
     </item>
    </layout>
   </item>
   <item row="0" column="1">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="spacing">
      <number>5</number>
     </property>
     <item>
      <widget class="QLabel" name="labTim_2">
       <property name="text">
        <string>历史信息</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="txtInfo"/>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="17,17,19">
     <property name="spacing">
      <number>10</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QGroupBox" name="fraUART">
       <property name="title">
        <string>XTend配置</string>
       </property>
       <widget class="QLabel" name="labUPort">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>40</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>串口</string>
        </property>
       </widget>
       <widget class="QLabel" name="labUBaud">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>60</y>
          <width>60</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>波特率</string>
        </property>
       </widget>
       <widget class="MyComboBox" name="cbxUPort">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>20</y>
          <width>100</width>
          <height>25</height>
         </rect>
        </property>
       </widget>
       <widget class="MyComboBox" name="cbxUBaud">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>60</y>
          <width>100</width>
          <height>25</height>
         </rect>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="btnURefresh">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>95</y>
          <width>70</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>刷新</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnUToggle">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>95</y>
          <width>70</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>打开</string>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="fraTcp">
       <property name="title">
        <string>TCP配置</string>
       </property>
       <widget class="QPushButton" name="btnTToggle">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>95</y>
          <width>70</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>监听</string>
        </property>
       </widget>
       <widget class="MyComboBox" name="cbxTIP">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>20</y>
          <width>131</width>
          <height>25</height>
         </rect>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="currentText">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="labTIP">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>60</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>IP</string>
        </property>
       </widget>
       <widget class="QLabel" name="labTPort">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>60</y>
          <width>60</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>端口号</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="txtTPort">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>60</y>
          <width>101</width>
          <height>25</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="btnTRefresh">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>95</y>
          <width>70</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>刷新</string>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="fraTcp_2">
       <property name="title">
        <string>文本操作</string>
       </property>
       <widget class="QPushButton" name="btnSend">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>20</y>
          <width>80</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>发送</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnFSend">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>80</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>发送文件</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnHClear">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>60</y>
          <width>80</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>清除历史</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnSClear">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>60</y>
          <width>80</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>清除发送</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnFSend_3">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>100</y>
          <width>80</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>发送文件</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnSend_3">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>100</y>
          <width>80</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>发送</string>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="fraLoRa">
     <property name="title">
      <string>LoRa配置</string>
     </property>
     <widget class="QLabel" name="labLPort">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>40</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>串口</string>
      </property>
     </widget>
     <widget class="QLabel" name="labLBaud">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>60</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>波特率</string>
      </property>
     </widget>
     <widget class="MyComboBox" name="cbxLPort">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>81</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="MyComboBox" name="cbxLBaud">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>60</y>
        <width>81</width>
        <height>25</height>
       </rect>
      </property>
      <property name="editable">
       <bool>true</bool>
      </property>
      <property name="currentText">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="btnLToggle">
      <property name="geometry">
       <rect>
        <x>75</x>
        <y>95</y>
        <width>60</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>打开</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtTim">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>60</y>
        <width>51</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="labTim">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>20</y>
        <width>51</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>倒计时</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnLListen">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>95</y>
        <width>60</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>监听</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnLRefresh">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>95</y>
        <width>60</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>刷新</string>
      </property>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WgtScroll</class>
   <extends>QWidget</extends>
   <header>wgtscroll.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MyComboBox</class>
   <extends>QComboBox</extends>
   <header>mycombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
